File: examples/triangle/worker.js
var cluster_lib = require('../../lib/cluster');

//Working computer

var cluster = cluster_lib.createCluster();
cluster.queues.push('working');

cluster.worker.on('something_long', function(args, respond_to, job_id) {
    //got a new task to do, lets answer
    cluster.answer(respond_to, 'something_long', job_id, ['Hello world!'],
        function(err, resp) { });
});

cluster.work_loop();
console.log('Worker is started', cluster.self());

File: examples/triangle/worker.rb
require 'json'
require 'redis'

REDIS = 'localhost:6379'

class Cluster

  def initialize
    @_client = {}
    @loop = true
  end

  def client server
    unless @_client.key? server
      hp = server.split(':')
      @_client[server] = Redis.new host: hp[0], port: hp[1].to_i
    end
    @_client[server]
  end

  def loop
    @loop = true
    while @loop
      task = client(REDIS).blpop 'working', 0
      if task
        queue = task[0]
        action = JSON.parse(task[1])
        args = action[1]
        answer = action[2]
        job_id = action[3]
        client(answer).something_long job_id, ['Hello from ruby'].to_json
      end
    end
  end

end

cluster = Cluster.new

cluster.loop
